### PR TITLE
MLR 323

### PR DESCRIPTION
**Pull request by cschroed-usgs**
Added the missing ingredient -- actually using the new CRU validator!
Rewrote CruValidator class and tests.
To improve test isolation, provided all end-to-end error validator tests with a fresh ErrorValidator instance.
Converted broken and/or irrelevant stationIx end-to-end tests into working, relevant tests.

### CODE REVIEW

File: mlrvalidator/schemas/error_schema.yml [hunk 1]
```diff
@@ -13,6 +13,11 @@ stationName:
     maxlength: 50
     is_empty: False
     regex: ^[^\t#\*\\"\^_\$]*$
```
Comment (cschroed-usgs) [inline mlrvalidator/schemas/error_schema.yml]: This should always be present now that these changes in the gateway have been merged:
https://github.com/USGS-CIDA/MLR-Gateway/pull/59

File: mlrvalidator/validators/tests/test_end_to_end_error_validator.py [hunk 2]
```diff
@@ -3,3635 +3,3728 @@
 
 from app import application
 from ..error_validator import ErrorValidator
+import requests_mock
```
Comment (cschroed-usgs) [inline mlrvalidator/validators/tests/test_end_to_end_error_validator.py]: This global variable was permitting the behavior of one test to have side effects on another test. This provoked inconsistent errors that could vary depending on the order test methods were run. To minimize confusion I switched all tests to instantiate their own ErrorValidator instead.

**Comment by cschroed-usgs** [comment]
Thanks! I added her.

File: mlrvalidator/validators/tests/test_end_to_end_error_validator.py [hunk 3]
```diff
@@ -3,3635 +3,3728 @@
...
 
-validator = ErrorValidator(application.config['SCHEMA_DIR'], application.config['REFERENCE_FILE_DIR'],
-                           'http://localhost')
 
+class BaseE2ETestCase(TestCase):
+    def setUp(self):
+        self.v = ErrorValidator(application.config['SCHEMA_DIR'], application.config['REFERENCE_FILE_DIR'],
```
Comment (mbucknell) [inline mlrvalidator/validators/tests/test_end_to_end_error_validator.py]: Something to think about if the testing time takes too long is to refactor ErrorValidator to pass in the data from the files that it needs.

Comment (cschroed-usgs) [reply mlrvalidator/validators/tests/test_end_to_end_error_validator.py]: Fosho! I created MLR-356 to address that. Please feel free to chime in on that ticket with additional suggestions.

**Pull request merged**